src/modules/airspeed_selector/airspeed_selector_main.cpp

```diff
@@ -540,19 +543,29 @@ void AirspeedModule::select_airspeed_and_publish()
 		}
 	}
 
-	/* publish critical message (and log) in index has changed */
-	/* Suppress log message if still on the ground and no airspeed sensor connected */
-	if (_valid_airspeed_index != _prev_airspeed_index && (_number_of_airspeed_sensors > 0
-			|| !_vehicle_land_detected.landed)) {
-		mavlink_log_critical(&_mavlink_log_pub, "Airspeed: switched from sensor %i to %i", _prev_airspeed_index,
-				     _valid_airspeed_index);
+	if (_valid_airspeed_index != _prev_airspeed_index) {
+		/* publish critical message (and log) in index has changed */
+		/* Suppress log message if still on the ground and no airspeed sensor connected */
+		if ((_number_of_airspeed_sensors > 0) || !_vehicle_land_detected.landed) {
+			mavlink_log_critical(&_mavlink_log_pub, "Airspeed: switched from sensor %i to %i", _prev_airspeed_index,
+					     _valid_airspeed_index);
+		}
+
+		if (_valid_airspeed_index >= 0 && _valid_airspeed_index <= MAX_NUM_AIRSPEED_SENSORS) {
+
+			for (auto &sub : _airspeed_subs) {
```

Comment:
I lost some context since the last review - what is the unregistration/registration of the airspeed_subs for here?

Reply:
This is so that `airspeed_selector` is scheduled to run immediately after the current primary airspeed publishes.

Reply:
aha, good :+1: